mod/wikirate_source/set/type/source/source.rb

```diff
@@ -88,12 +85,32 @@ def has_file_or_text?
   
 end
 
+def is_url? url
+  url.start_with?("http://") || url.start_with?("https://")
+end
+
+def is_wikirate_url? url
+  wikirate_url = "#{ Card::Env[:protocol] }#{ Card::Env[:host] }"
+  url.start_with?(wikirate_url)
+end
+
+def get_card url
+  if is_wikirate_url?(url)
+    # try to convert the link to source card, 
+    # easier for users to add source in +source editor
+    uri = URI.parse(URI.unescape(url))
+    Card[uri.path]
+  else
+    Card[url]
+  end
+end
+
 def download_file_and_add_to_plus_file url
-  url.gsub!(/ /, '%20')  
-  subcards["+File"] = {
-    :remote_file_url=>url,:type_id=>Card::FileID,:content=>"dummy"
-  }
-  subcards.delete("+#{ Card[:wikirate_link].name }")
+  url.gsub!(/ /, '%20')
+  add_subcard '+file', remote_file_url: url, type_id: Card::FileID,
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.